stacklight_tests/tests/prometheus/test_alerts.py

```diff
@@ -144,3 +144,100 @@ def test_es_health_alert(self, cluster, prometheus_alerting):
         del(criteria["severity"])
         prometheus_alerting.check_alert_status(
             criteria, is_fired=False, timeout=6 * 60)
+
+
+class TestNeutronAlerts(object):
+    def test_neutron_api_alert(self, destructive, cluster,
+                               prometheus_alerting):
+        service = "neutron-server"
+        controllers = cluster.get_controllers()
+        criteria = {
+            "name": "NeutronAPIDown",
+        }
+        prometheus_alerting.check_alert_status(criteria, is_fired=False)
+        criteria.update({"severity": "down"})
+        logger.info("Stop {} service on {} nodes".format(
+            service, ', '.join([str(ctl.hostname) for ctl in controllers])))
+        for controller in controllers:
+            destructive.append(
+                lambda: controller.os.manage_service(service, "start"))
+            controller.os.manage_service(service, "stop")
+        prometheus_alerting.check_alert_status(
+            criteria, is_fired=True, timeout=10 * 60)
+        for controller in controllers:
+            controller.os.manage_service(service, "start")
+        del(criteria["severity"])
+        prometheus_alerting.check_alert_status(
+            criteria, is_fired=False, timeout=10 * 60)
+
+    def test_neutron_all_agents_down_alert(self, cluster, destructive,
+                                           prometheus_alerting):
+        service = "neutron-l3-agent"
+        hosts = [host for host in cluster.hosts if host.os.check_process(
+            service)]
+        criteria = {
+            "name": "NeutronAllAgentsDown",
+            "service": service,
+        }
+        prometheus_alerting.check_alert_status(criteria, is_fired=False)
+        criteria.update({"severity": "down"})
+        logger.info("Stop {} service on {} nodes".format(
+            service, ', '.join([str(h.hostname) for h in hosts])))
+        for host in hosts:
+            destructive.append(
+                lambda: host.os.manage_service(service, "start"))
+            host.os.manage_service(service, "stop")
+        prometheus_alerting.check_alert_status(
+            criteria, is_fired=True, timeout=10 * 60)
+        for host in hosts:
+            host.os.manage_service(service, "start")
+        del(criteria["severity"])
```

Comment:
I suppose it would be better to change check_alert_status to ignore additional fields